### DISCUSSION TITLE
Unable to get correct response from AI

### DISCUSSION THREAD

**Discussion by Manish-22** [microsoft/semantic-kernel#9044] [Q&A] [answered]
I'm not sure if this is a semantic kernel issue but when an user asks 

> How can I change the state of light?"

for the function
```
   [KernelFunction("change_state")]
   [Description("Changes the state of the light")]
   [return: Description("The updated state of the light; will return null if the light does not exist")]
   public async Task<LightModel?> ChangeStateAsync(int id, bool isOn)
   {
      var light = lights.FirstOrDefault(light => light.Id == id);

      if (light == null)
      {
         return null;
      }

      // Update the light with the new state
      light.IsOn = isOn;

      return light;
   }
```
The AI responds 

> To change the state of light you can use the change_state funtion. You need to provide the id of the light and the state you want the light to be in.
> for example 
> const {result} = await ai.call<funtions.LightsPlugin.change_state>('LightsPlugin.change_state', .....) 

I don't want the user to see the internal working of the funtion. The AI should ideally respond with just "To change the state of the light you need to provide the id of the light and the state you want the light to be in.". 

Another observation is that when the user asks "I want to change the state of the light" the AI gives the expected response back. I reckon it is the "how" that is throwing the AI off.

Any modfication that should be made in the system prompt for this to be achieved?

**Comment by evchaki** [comment, marked as answer]
You should be able to do this with system prompts to tell the user the way to respond.  For your ideal state "To change the state of the light you need to provide the id of the light and the state you want the light to be in." the function would know it needs an ID as input.  You can do examples in the system prompt on how you want it to respond to the user.
